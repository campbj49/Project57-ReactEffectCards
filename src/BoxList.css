.BoxList-main {
  display: flex;
  align-items: center;
  justify-content: center;
}

.BoxList-main > i {
  cursor: pointer;
  margin: 10px;
  color: #aaa;
}

.BoxList-main > i:hover {
  color: black;
}

.bi {
  font-size: 35px
}
