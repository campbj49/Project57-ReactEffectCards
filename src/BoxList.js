import { useState } from "react";
import "./BoxList.css";
import Box from "./Box";
import NewBoxForm from "./NewBoxForm";

/** BoxList: displays list of created boxes and form for the creation of new boxes
 * 
 * State:
 * - boxList : list of all the boxes to be rendered
 * 
 * App --> BoxList --> Box
 *          |__> NewBoxForm
 */
 function BoxList(props) {
  const [list, setList] = useState([]);

  //function that removes the tartgeted box from the list
  function remove(e){
    setList(list.filter((l,i)=> e.target.id != i))
  }

  //function for the newly created box
  function onSubmit(evt){
    evt.preventDefault();
    let rawData = evt.target.querySelectorAll("input");
    let newBox = {
      height:parseInt(rawData[0].value),
      width: parseInt(rawData[1].value),
      backgroundColor: rawData[2].value
    }
    addList(newBox);
  }
  
  //function for adding a box to the list
  function addList(box){
    console.log(box);
    setList([...list, box]);
  }
  return (
    <div className="BoxList">
      <NewBoxForm onSubmit={onSubmit}/>
      {list.map((box,i) => 
        //console.log(box.heigh);
        <Box 
        height={box.height} 
        width={box.width}
        backgroundColor={box.backgroundColor}
        key={i}
        id={i}
        remove={remove}/>)}
    </div>
  );
}

export default BoxList;
