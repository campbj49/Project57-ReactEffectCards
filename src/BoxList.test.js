import { render, fireEvent } from "@testing-library/react";
import BoxList from "./BoxList";
import TEST_IMAGES from "./_testCommon.js";

//smoke test to make sure it renders in the first place from smoke test example
it("renders without crashing", function() {
  render(<BoxList/>);
});

// snapshot test from snapshot example
it("matches snapshot", function() {
  const {asFragment} = render(<BoxList/>);
  expect(asFragment()).toMatchSnapshot();
});

it("adding a box works", function() {
  const { container } = render(<BoxList/>);

  // expect there to be no boxes rendered yet
  expect(
    container.querySelector(".Box")
  ).not.toBeInTheDocument();

  //fill out the form and submit it
  container.querySelectorAll("input")[0].value = 500;
  container.querySelectorAll("input")[1].value = 250;
  container.querySelectorAll("input")[2].value = "blue";
  const subButton = container.querySelector("button");
  fireEvent.click(subButton);

  // expect a box to have appeared
  expect(
    container.querySelector(".Box")
  ).toBeInTheDocument();

  //click X

  const delButton = container.querySelector("p");
  fireEvent.click(delButton);

  // expect there to be no boxes rendered 
  expect(
    container.querySelector(".Box")
  ).not.toBeInTheDocument();
});

