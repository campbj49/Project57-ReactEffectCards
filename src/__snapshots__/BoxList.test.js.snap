// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="BoxList"
  >
    <form>
      <label
        for="height"
      >
        Height
      </label>
      <input
        id="height"
        required=""
        type="number"
      />
      <br />
      <label
        for="width"
      >
        Width
      </label>
      <input
        id="width"
        required=""
        type="number"
      />
      <br />
      <label
        for="backgroundColor"
      >
        Background Color
      </label>
      <input
        id="backgroundColor"
        required=""
        type="text"
      />
      <br />
      <button>
        Create Box
      </button>
    </form>
  </div>
</DocumentFragment>
`;
